tests/listusers/listusers.go

```diff
@@ -0,0 +1,220 @@
+// Package listusers contains integration tests for the ListUsers and StreamedListUsers APIs.
+package listusers
+
+import (
+	"context"
+	"math"
+	"testing"
+
+	oldparser "github.com/craigpastro/openfga-dsl-parser/v2"
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	parser "github.com/openfga/language/pkg/go/transformer"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/status"
+	"sigs.k8s.io/yaml"
+
+	listuserstest "github.com/openfga/openfga/internal/test/listusers"
+
+	"github.com/openfga/openfga/assets"
+	"github.com/openfga/openfga/pkg/tuple"
+	"github.com/openfga/openfga/pkg/typesystem"
+	"github.com/openfga/openfga/tests/check"
+)
+
+var writeMaxChunkSize = 40 // chunk write requests into a chunks of this max size
+
+type individualTest struct {
+	Name   string
+	Stages []*stage
+}
+
+type listUsersTests struct {
+	Tests []individualTest
+}
+
+type testParams struct {
+	schemaVersion string
+	client        ClientInterface
+}
+
+type stage struct {
+	Model               string
+	Tuples              []*openfgav1.TupleKey
+	ListUsersAssertions []*listuserstest.Assertion `json:"listUsersAssertions"`
+}
+
+type ClientInterface interface {
+	check.ClientInterface
+	ListUsers(ctx context.Context, in *openfgav1.ListUsersRequest, opts ...grpc.CallOption) (*openfgav1.ListUsersResponse, error)
+}
+
+// RunAllTests will invoke all list users tests
+func RunAllTests(t *testing.T, client ClientInterface) {
+	t.Run("RunAll", func(t *testing.T) {
+		t.Run("ListUsers", func(t *testing.T) {
+			t.Parallel()
+			testListUsers(t, client)
+		})
+	})
+}
+
+func testListUsers(t *testing.T, client ClientInterface) {
+	t.Run("Schema1_1", func(t *testing.T) {
+		t.Parallel()
+		runSchema1_1ListUsersTests(t, client)
+	})
+}
+
+func runSchema1_1ListUsersTests(t *testing.T, client ClientInterface) {
+	runTests(t, testParams{typesystem.SchemaVersion1_1, client})
+}
+
+func runTests(t *testing.T, params testParams) {
+	files := []string{
+		"tests/consolidated_1_1_tests.yaml",
+		//"tests/abac_tests.yaml", TODO add back
+	}
+
+	var allTestCases []individualTest
+
+	for _, file := range files {
+		var b []byte
+		var err error
+		schemaVersion := params.schemaVersion
+		if schemaVersion == typesystem.SchemaVersion1_1 {
+			b, err = assets.EmbedTests.ReadFile(file)
+		}
+		require.NoError(t, err)
+
+		var testCases listUsersTests
+		err = yaml.Unmarshal(b, &testCases)
+		require.NoError(t, err)
+
+		allTestCases = append(allTestCases, testCases.Tests...)
+	}
+
+	for _, test := range allTestCases {
+		test := test
+		runTest(t, test, params, false)
+		// TODO uncomment
+		// runTest(t, test, params, true)
```

Comment:
Same question here. My preference is to uncomment in this PR rather than having lingering TODOs.

Reply:
But then how will I remember that I need to add this back?

Reply:
Ticket. Todos don't remind us to do anything, shown by the numerous TODOs we have in the codebase today 😄

Reply:
OK, created tickets 👍